File: JavaScript/Problem003.js
const solution = (num) => {
  let factor = 0
  let div = 2
  while (div <= num) {
    if (num % div == 0) {
      factor = div
      num /= div
    } else {
      ++div
    }
  }
  return factor
}

console.log(solution(600851475143))

File: Python/Problem003.py
def solution(n):
    factors = []
    div = 2
    while (div <= n):
        if n % div == 0:
            factors.append(div)
            n /= div
        else:
            div = div + 1
    return factors[len(factors) - 1]

print (solution(25))
